File: _experimental/_scala/numbers/src/main/scala/numbers_recursive.scala
class NumbersRecursive {
  def fibonacci(n: Int): Int = {
    if (n <= 0) 0
    else if (n == 1) 1
    else fibonacci(n - 1) + fibonacci(n - 2)
  }

  def factorial(n: Int): Int = {
    if (n <= 1) 1
    else n * factorial(n - 1)
  }

  def sum_numbers(n: Int): Int = {
    if (n <= 0) 0
    else n + sum_numbers(n - 1)
  }
}

File: _experimental/_lua/numbers/src/numbers_recursive.lua
local numbers_recursive = {}

function numbers_recursive.fibonacci(n)
  if n <= 0 then
    return 0
  elseif n == 1 then
    return 1
  else
    return numbers_recursive.fibonacci(n - 1) + numbers_recursive.fibonacci(n - 2)
  end
end

function numbers_recursive.factorial(n)
  if n == 0 then
    return 1
  else
    return n * numbers_recursive.factorial(n - 1)
  end
end

function numbers_recursive.sum_numbers(n)
  if n <= 0 then
    return 0
  else
    return n + numbers_recursive.sum_numbers(n - 1)
  end
end
return numbers_recursive